### PR TITLE
initial port to python 3

### PR DESCRIPTION
**Pull request by jordic**

### CODE REVIEW

**Comment by icemac** [comment]
I converted this PR into a draft. Before it can be reviewed CI integration has to be activated for this repository. This requires a `.travis.yml`. Feel free to create a new PR for CI integration – sorry I have no time and energy to do this myself, but I am willing to help where admin rights are needed.

Additionally please note the following:

According to the contributing policies of the zopefoundation organization you need to sign a contributor agreement before any non-trivial change before it can be merged. For details please consult the [Contributing guidelines for zopefoundation projects](https://github.com/zopefoundation/.github/blob/master/CONTRIBUTING.md).

**Comment by jordic** [comment]
> According to the contributing policies of the zopefoundation organization you need to sign a contributor agreement before any non-trivial change before it can be merged. For details please consult the [Contributing guidelines for zopefoundation projects](https://github.com/zopefoundation/.github/blob/master/CONTRIBUTING.md).

There's no problem with this, but I'm also member and contributor on the plone foundation.

**Comment by dataflake** [comment]
@jordic The forms to sign are the same, but it is still a requirement to sign it separately for the `zopefoundation` repositories.

**Comment by jordic** [comment]
I'm using it on zope5 (but it's just an experiment)... will see if it evolves to something.

**Comment by jordic** [comment]
> According to the contributing policies of the zopefoundation organization you need to sign a contributor agreement before any non-trivial change before it can be merged. For details please consult the [Contributing guidelines for zopefoundation projects](https://github.com/zopefoundation/.github/blob/master/CONTRIBUTING.md).

Form signed and sent to agreements@plone.org

File: Products/SQLAlchemyDA/__init__.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 
 def initialize(context):
-    from da import (SAWrapper, manage_addSAWrapper,
+    from Products.SQLAlchemyDA.da import (SAWrapper, manage_addSAWrapper,
```
Comment (dataflake) [inline Products/SQLAlchemyDA/__init__.py:19]: Make this a relative import

File: Products/SQLAlchemyDA/da.py [hunk 2]
```diff
@@ -300,7 +300,8 @@ def getInfo(self):
         if wrapper is not None:
             d = self.sa_zope_wrapper().__dict__.copy()
             d['DSN'] = self.sa_zope_wrapper().dsn
-            for k in d.keys()[:]:
+            dd = d.copy()
+            for k in dd.keys():
```
Comment (dataflake) [inline Products/SQLAlchemyDA/da.py:304]: Why is this needed?

File: Products/SQLAlchemyDA/da.py [hunk 3]
```diff
@@ -451,10 +452,10 @@ def manage_doQuery(self, query):
...
     def manage_formatItem(self, s):
         """ used by query form """
-        if isinstance(s, unicode):
-            return s
-        if isinstance(s, str):
-            return unicode(s, self.encoding, 'ignore')
+        # if isinstance(s, unicode):
+        #    return s
+        # if isinstance(s, str):
+        #    return unicode(s, self.encoding, 'ignore')
```
Comment (dataflake) [inline Products/SQLAlchemyDA/da.py:458]: The intent here was to leave unicode strings untouched and convert binary strings to unicode in a safe way. Removing all that invites `UnicodeDecodeError` on the subsequent `str` call.

**Review by dataflake — CHANGES_REQUESTED** [review]

File: Products/SQLAlchemyDA/da.py (hunk 2 above)
Comment (jordic) [reply Products/SQLAlchemyDA/da.py:304]: the old one is cloning the list of keys. On python3.8 (at least) keys is of type dict_keys(['a']), and 'dict_keys' object is not subscriptable. That's because keys on dict couldn't be mutated while iterating..
